src/org/wikipedia/Wiki.java

```diff
@@ -4919,8 +4918,8 @@ public boolean isWatched(String title) throws IOException, CredentialNotFoundExc
                 boolean iscat = namespace(member) == CATEGORY_NAMESPACE;
                 if (maxdepth > 0 && iscat && !visitedcategories.contains(member))
                 {
-                    String[] categoryMembers = getCategoryMembers(member, --maxdepth, visitedcategories, ns);
                     visitedcategories.add(member);
+                    String[] categoryMembers = getCategoryMembers(member, --maxdepth, visitedcategories, ns);
```

Comment:
Urgs sorry, I didn’t take the time I should have taken to process your comment. You’re right of course, the decrement should happen after the do-while-loop. If you want, I can have another look this afternoon and prepare another patch. Or you can fix it yourself.